src/main/java/command/Command.java

```diff
@@ -0,0 +1,22 @@
+package command;
+
+import exception.DukeException;
+import logic.Storage;
+import logic.Ui;
+import tasks.TaskList;
+
+/**
+ * Abstract class of all the commands that is created
+ * after parsing the user's input
+ */
+
+public abstract class Command {
+
+    public final String input;
+
+    public Command(String input) {
+        this.input = input;
+    }
+    public abstract void execute(TaskList tasks, Ui ui, Storage storage) throws DukeException;
+    public abstract boolean isExit();
```

Comment:
Perhaps instead of making this an abstract method, you could have a default implementation that returns `false`? That way you minimize code duplication across the different subclasses of `Command` and you'll only have to override this implementation for subclasses in which you want to return `true`